server/src/test/java/com/adobe/testing/s3mock/its/AmazonClientUploadIT.java

```diff
@@ -707,4 +707,64 @@ public void shouldAddAndRetrieveTags() {
     assertThat("The vaule of the tag placed did not match",
         getObjectTaggingResult.getTagSet().get(0).getValue(), is("bar"));
   }
+
+  /**
+   * Creates a bucket, stores a file, get files with eTag requrements.
+   */
+  @Test
+  public void shouldCreateAndRespectEtag() throws Exception {
+    final File uploadFile = new File(UPLOAD_FILE_NAME);
+
+    s3Client.createBucket(BUCKET_NAME);
+    PutObjectResult returnObj = s3Client.putObject(
+            new PutObjectRequest(BUCKET_NAME,
+                    uploadFile.getName(),
+                    uploadFile));
+
+    // wit eTag
+    GetObjectRequest requestWithEtag = new GetObjectRequest(BUCKET_NAME, uploadFile.getName());
+    requestWithEtag.setMatchingETagConstraints(new ArrayList<String>() { {
```

Comment:
please use `java.util.Collections#singletonList(...)` here. Same for other `setNonmatchingETagConstraints` resp. `setMatchingETagConstraints` invocations